snapcraft/tests/test_plugin_qmake.py

```diff
@@ -0,0 +1,169 @@
+                         .format(options_items_type))
+
+        # Check install-via property
+        qt_version = properties['qt-version']
+        self.assertTrue('enum' in qt_version,
+                        'Expected "enum" to be included in "qt_version"')
+        self.assertFalse('default' in qt_version,
+                         '"default" was unexpectedly included in "qt_version"')
+
+        qt_version_enum = qt_version['enum']
+        # Using sets for order independence in the comparison
+        self.assertEqual(set(['qt4', 'qt5']), set(qt_version_enum))
+
+        self.assertTrue('build-properties' in schema,
+                        'Expected schema to include "build-properties"')
+        build_properties = schema['build-properties']
+        self.assertTrue('options' in build_properties)
+
+        self.assertTrue('pull-properties' in schema,
+                        'Expected schema to include "pull-properties"')
+        pull_properties = schema['pull-properties']
+        self.assertTrue('qt-version' in pull_properties)
+
+        # Finally, verify that qt-version is required
+        required = schema['required']
+        self.assertTrue('qt-version' in required,
+                        'Expected "qt-version" to be required')
+
+    def test_build_referencing_sourcedir_if_no_subdir(self):
+        plugin = qmake.QmakePlugin('test-part', self.options,
+                                   self.project_options)
+        os.makedirs(plugin.builddir)
+        plugin.build()
+
+        self.run_mock.assert_has_calls([
+            mock.call(['qmake', plugin.sourcedir], cwd=plugin.builddir,
+                      env=mock.ANY),
+            mock.call(['make', '-j2'], cwd=plugin.builddir, env=mock.ANY),
+            mock.call(['make', 'install',
+                       'INSTALL_ROOT={}'.format(plugin.installdir)],
+                      cwd=plugin.builddir, env=mock.ANY)])
+
+    def test_build_referencing_sourcedir_with_subdir(self):
+        self.options.source_subdir = 'subdir'
+
+        plugin = qmake.QmakePlugin('test-part', self.options,
+                                   self.project_options)
+        os.makedirs(plugin.builddir)
+        plugin.build()
+
+        sourcedir = os.path.join(
+            plugin.sourcedir, plugin.options.source_subdir)
+        self.run_mock.assert_has_calls([
+            mock.call(['qmake', sourcedir], cwd=plugin.builddir, env=mock.ANY),
+            mock.call(['make', '-j2'], cwd=plugin.builddir, env=mock.ANY),
+            mock.call(['make', 'install',
+                       'INSTALL_ROOT={}'.format(plugin.installdir)],
+                      cwd=plugin.builddir, env=mock.ANY)])
+
+    def test_build_environment(self):
+        plugin = qmake.QmakePlugin('test-part', self.options,
+                                   self.project_options)
+        os.makedirs(plugin.builddir)
+        plugin.build()
+
+        expected = {'QT_SELECT': 'qt5'}
```

Comment:
no test for qt4? You can make one test, with two subTests:
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests